bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/Activator.java

```diff
@@ -71,7 +72,11 @@ public Object addingService(ServiceReference reference) {
                 StorageService storage = (StorageService) bc.getService(reference);
                 if (storage != null) {
                     final ManagedRuleProvider rp = new ManagedRuleProvider(storage);
-                    ruleRegistry = new RuleRegistryImpl(rm, rp);
+
+                    Storage storageDisabledRules = storage.getStorage("automation_rules_disabled",
+                            this.getClass().getClassLoader());
+
+                    ruleRegistry = new RuleRegistryImpl(rm, rp, storageDisabledRules);
```

Comment:
That with the local storage is fine, but the ruleRegistry will only be created if a StorageService is added. This causes that the RuleRegistry can not live without a storageService. In my opinion it would be better to track the StorageService in the RuleRegistry itself or add setter/unsetter methods for it which could be called here. Additionally in the setter there should be a synchronization of the inMemory data and the storage-provided Data.